x-pack/plugins/observability/public/pages/alerts/components/alerts_search_bar.tsx

```diff
@@ -35,20 +35,11 @@ export function AlertsSearchBar({
     return new TimeHistory(new Storage(localStorage));
   }, []);
   const [queryLanguage, setQueryLanguage] = useState<QueryLanguageType>('kuery');
-
-  const compatibleIndexPatterns = useMemo(
-    () =>
-      dynamicIndexPatterns.map((dynamicIndexPattern) => ({
-        title: dynamicIndexPattern.title ?? '',
-        id: dynamicIndexPattern.id ?? '',
-        fields: dynamicIndexPattern.fields,
-      })),
-    [dynamicIndexPatterns]
-  );
+  const { value: dataView, loading, error } = useAlertDataView(featureIds);
 
   return (
     <SearchBar
-      indexPatterns={compatibleIndexPatterns as DataView[]}
+      indexPatterns={loading || error ? undefined : [dataView!]}
```

Comment:
Could we think of a better way to handle the error and the loading? Because if we pass `undefined` it will break the page aka no Search bar.
<img width="1960" alt="Screenshot 2022-10-04 at 12 14 18" src="https://user-images.githubusercontent.com/6838659/193794433-d31f471a-fcde-4828-9e38-47f50975af9f.png">

```suggestion
      indexPatterns={loading || error ? [] : [dataView!]}
```
Passing an empty array will keep the Search bar functional but without the data view.

Reply:
Interesting, when I checked the type of `indexPatterns` I saw `SearchBarOwnProps<QT>.indexPatterns?: DataView[] | undefined` so it seemed to me that `SearchBar` can handle `undefined` indexPatterns.
I will change this part.

One improvement can be to change the logic relate to `indexPatterns` in the search bar and 
1. either not allowing `undefined`
2. or setting `[]` as the default value for `indexPatterns`